### ISSUE TITLE
Building port audio fails on macOS Mojave for macOS target

### ISSUE THREAD

**Issue by BusFactor1Inc** [part-cw/lambdanative#223]
Attempting to build uSquish for macOS on Mojave fails with the following error on port audio:

```
=> libportaudio..
 => downloading http://www.portaudio.com/archives/pa_stable_v19_20140130.tgz..
 == hash 526a7955de59016a06680ac24209ecb6ce05527d
 => extracting /Users/burton/Library/Caches/lambdanative/packages/pa_stable_v19_20140130.tgz..
 => patching source...
 => applying patches from ../macos.patch
 => applying patches from ../openbsd.patch
 => configuring source..
...
configure: error: Couldn't find 10.5 to 10.13 SDK
ERROR: failed with exit code 64
>> configure failed
BUILD FAILED - configure with verbose option for more information
make: *** [all] Error 1
```

**Comment by mgorges** [comment]
You might have to touch the `libraries/libportaudio/make.sh` to force the rebuild.

**Comment by BusFactor1Inc** [comment]
Still the same error.

**Comment by mgorges** [comment]
Did you touch the file after pulling? (I just tested this on MacOS 10.14 with Xcode 10.1.)

**Comment by BusFactor1Inc** [comment]
Sorry, I misunderstood and forgot to pull first.  My mistake.

**Comment by ddunsmuir** [comment]
I am having this same issue on High Sierra with XCode 9.4.1, and pulling and touching that file did not work. I also tried a make scrub.

**Comment by ddunsmuir** [comment]
Here's the output in verbose mode from => libportaudio onward:

 => libportaudio..
  => shasum.. ok
 => extracting /Users/DTD/Documents/BCCH/Phone/lambdanative-cache/packages/pa_stable_v190600_20161030.tgz..
  => tar.. ok
 => patching source...
 => applying patches from ../macos.patch
patch -p0 --ignore-whitespace < ../macos.patch
patching file configure.in
 => applying patches from ../openbsd.patch
patch -p0 --ignore-whitespace < ../openbsd.patch
patching file Makefile.in
patching file configure.in
Hunk #3 succeeded at 420 (offset 61 lines).
Hunk #4 succeeded at 530 (offset 61 lines).
patching file include/portaudio.h
Hunk #1 succeeded at 287 (offset 51 lines).
patching file portaudio-2.0.pc.in
patching file src/os/unix/pa_unix_hostapis.c
patching file src/os/unix/pa_unix_util.c
patching file src/hostapi/sndio/pa_sndio.c
autom4te: need GNU m4 1.4 or later: /opt/local/bin/gm4
 => configuring source..
  CHOST=x86_64-apple-darwin17.7.0   PKG_CONFIG_PATH=/Users/DTD/Documents/BCCH/Phone/lambdanative-cache/macosx/lib/pkgconfig   PATH="/Users/DTD/Documents/BCCH/Phone/lambdanative-cache/macosx/bin:.:/opt/local/bin:/opt/local/sbin:/usr/bin:/bin:/usr/sbin:/sbin:/usr/local/bin:/opt/X11/bin:/usr/local/android-sdk-r25.2.5/tools:/usr/local/android-sdk-r25.2.5/platform-tools"   LDFLAGS=-L/Users/DTD/Documents/BCCH/Phone/lambdanative-cache/macosx/lib   LD_LIBRARY_PATH=/Users/DTD/Documents/BCCH/Phone/lambdanative-cache/macosx/lib   CPPFLAGS=-I/Users/DTD/Documents/BCCH/Phone/lambdanative-cache/macosx/include   CC="gcc -g -O0 -DMACOSX -m64  -I/Users/DTD/Documents/BCCH/Phone/lambdanative-cache/macosx/include -L/Users/DTD/Documents/BCCH/Phone/lambdanative-cache/macosx/lib"   AR="ar"   RANLIB="ranlib"   NM="nm"   LD="ld"   AS=""   CPP=""   OBJCOPY=""   STRIP="strip"   GPROF=""   READELF=""   OBJDUMP=""   ./configure --prefix=/Users/DTD/Documents/BCCH/Phone/lambdanative-cache/macosx --enable-static --disable-shared --without-jack 
checking build system type... x86_64-apple-darwin17.7.0
checking host system type... x86_64-apple-darwin17.7.0
checking target system type... x86_64-apple-darwin17.7.0
checking for gcc... gcc -g -O0 -DMACOSX -m64 -I/Users/DTD/Documents/BCCH/Phone/lambdanative-cache/macosx/include -L/Users/DTD/Documents/BCCH/Phone/lambdanative-cache/macosx/lib
checking whether the C compiler works... yes
checking for C compiler default output file name... a.out
checking for suffix of executables... 
checking whether we are cross compiling... no
checking for suffix of object files... o
checking whether we are using the GNU C compiler... yes
checking whether gcc -g -O0 -DMACOSX -m64 -I/Users/DTD/Documents/BCCH/Phone/lambdanative-cache/macosx/include -L/Users/DTD/Documents/BCCH/Phone/lambdanative-cache/macosx/lib accepts -g... yes
checking for gcc -g -O0 -DMACOSX -m64 -I/Users/DTD/Documents/BCCH/Phone/lambdanative-cache/macosx/include -L/Users/DTD/Documents/BCCH/Phone/lambdanative-cache/macosx/lib option to accept ISO C89... none needed
checking how to print strings... printf
checking for a sed that does not truncate output... /usr/bin/sed
checking for grep that handles long lines and -e... /usr/bin/grep
checking for egrep... /usr/bin/grep -E
checking for fgrep... /usr/bin/grep -F
checking for ld used by gcc -g -O0 -DMACOSX -m64 -I/Users/DTD/Documents/BCCH/Phone/lambdanative-cache/macosx/include -L/Users/DTD/Documents/BCCH/Phone/lambdanative-cache/macosx/lib... ld
checking if the linker (ld) is GNU ld... no
checking for BSD- or MS-compatible name lister (nm)... nm
checking the name lister (nm) interface... rm: conftest.dSYM: is a directory
BSD nm
checking whether ln -s works... yes
checking the maximum length of command line arguments... 196608
checking whether the shell understands some XSI constructs... yes
checking whether the shell understands "+="... yes
checking how to convert x86_64-apple-darwin17.7.0 file names to x86_64-apple-darwin17.7.0 format... func_convert_file_noop
checking how to convert x86_64-apple-darwin17.7.0 file names to toolchain format... func_convert_file_noop
checking for ld option to reload object files... -r
checking for objdump... objdump
checking how to recognize dependent libraries... pass_all
checking for dlltool... dlltool
checking how to associate runtime and link libraries... printf %s\n
checking for archiver @FILE support... rm: conftest.dSYM: is a directory
no
checking for strip... strip
checking for ranlib... ranlib
checking for gawk... no
checking for mawk... no
checking for nawk... no
checking for awk... awk
checking command to parse nm output from gcc -g -O0 -DMACOSX -m64 -I/Users/DTD/Documents/BCCH/Phone/lambdanative-cache/macosx/include -L/Users/DTD/Documents/BCCH/Phone/lambdanative-cache/macosx/lib object... rm: conftest.dSYM: is a directory
ok
checking for sysroot... no
checking for mt... no
checking if : is a manifest tool... no
checking for dsymutil... dsymutil
checking for nmedit... nmedit
checking for lipo... lipo
checking for otool... otool
checking for otool64... no
checking for -single_module linker flag... yes
checking for -exported_symbols_list linker flag... yes
checking for -force_load linker flag... yes
checking how to run the C preprocessor... gcc -g -O0 -DMACOSX -m64 -I/Users/DTD/Documents/BCCH/Phone/lambdanative-cache/macosx/include -L/Users/DTD/Documents/BCCH/Phone/lambdanative-cache/macosx/lib -E
checking for ANSI C header files... yes
checking for sys/types.h... yes
checking for sys/stat.h... yes
checking for stdlib.h... yes
checking for string.h... yes
checking for memory.h... yes
checking for strings.h... yes
checking for inttypes.h... yes
checking for stdint.h... yes
checking for unistd.h... yes
checking for dlfcn.h... yes
checking for objdir... .libs
checking if gcc -g -O0 -DMACOSX -m64 -I/Users/DTD/Documents/BCCH/Phone/lambdanative-cache/macosx/include -L/Users/DTD/Documents/BCCH/Phone/lambdanative-cache/macosx/lib supports -fno-rtti -fno-exceptions... yes
checking for gcc -g -O0 -DMACOSX -m64 -I/Users/DTD/Documents/BCCH/Phone/lambdanative-cache/macosx/include -L/Users/DTD/Documents/BCCH/Phone/lambdanative-cache/macosx/lib option to produce PIC... -fno-common -DPIC
checking if gcc -g -O0 -DMACOSX -m64 -I/Users/DTD/Documents/BCCH/Phone/lambdanative-cache/macosx/include -L/Users/DTD/Documents/BCCH/Phone/lambdanative-cache/macosx/lib PIC flag -fno-common -DPIC works... yes
checking if gcc -g -O0 -DMACOSX -m64 -I/Users/DTD/Documents/BCCH/Phone/lambdanative-cache/macosx/include -L/Users/DTD/Documents/BCCH/Phone/lambdanative-cache/macosx/lib static flag -static works... no
checking if gcc -g -O0 -DMACOSX -m64 -I/Users/DTD/Documents/BCCH/Phone/lambdanative-cache/macosx/include -L/Users/DTD/Documents/BCCH/Phone/lambdanative-cache/macosx/lib supports -c -o file.o... yes
checking if gcc -g -O0 -DMACOSX -m64 -I/Users/DTD/Documents/BCCH/Phone/lambdanative-cache/macosx/include -L/Users/DTD/Documents/BCCH/Phone/lambdanative-cache/macosx/lib supports -c -o file.o... (cached) yes
checking whether the gcc -g -O0 -DMACOSX -m64 -I/Users/DTD/Documents/BCCH/Phone/lambdanative-cache/macosx/include -L/Users/DTD/Documents/BCCH/Phone/lambdanative-cache/macosx/lib linker (ld) supports shared libraries... yes
checking dynamic linker characteristics... darwin17.7.0 dyld
checking how to hardcode library paths into programs... immediate
checking whether stripping libraries is possible... yes
checking if libtool supports shared libraries... yes
checking whether to build shared libraries... no
checking whether to build static libraries... yes
checking for a BSD-compatible install... /usr/bin/install -c
checking whether ln -s works... yes
checking for ar... /opt/local/bin/ar
checking whether byte ordering is bigendian... no
checking for snd_pcm_open in -lasound... no
checking for HPI_SubSysCreate in -lhpi... no
checking sys/soundcard.h usability... no
checking sys/soundcard.h presence... no
checking for sys/soundcard.h... no
checking linux/soundcard.h usability... no
checking linux/soundcard.h presence... no
checking for linux/soundcard.h... no
checking machine/soundcard.h usability... no
checking machine/soundcard.h presence... no
checking for machine/soundcard.h... no
checking size of short... 2
checking size of int... 4
checking size of long... 8
checking for clock_gettime in -lrt... no
checking for clock_gettime... yes
checking for nanosleep... yes
configure: error: Could not find 10.5 to 10.12 SDK.
ERROR: failed with exit code 64
>> configure failed
BUILD FAILED
make: *** [all] Error 1
dustins-mbp:lambdanative DTD$

**Comment by mgorges** [comment]
You can't get this message: `configure: error: Could not find 10.5 to 10.12 SDK.` if you are on a current version of LambdaNative unless the `libraries/libportaudio/macos.patch` didn't get updated for some reason.

**Comment by mgorges** [comment]
A critical error message in your case is actually **autom4te: need GNU m4 1.4 or later: /opt/local/bin/gm4**, which indicates that `autoconf` failed; thus the patch to `configure.in` never got applied to `configure` and you end up with the error message.

**Comment by ddunsmuir** [comment]
I was able to solve this by updating m4. The latest version (available from brew) is 1.4.18. I also had to update my autom4te to point to the brew installed version instead of an old gm4.

For me, autom4te is in /opt/local/bin and I edited one line,
my $m4 = $ENV{"M4"} || '/opt/local/bin/gm4';
changed to
my $m4 = $ENV{"M4"} || '/usr/local/Cellar/m4/1.4.18/bin/m4';

**Comment by paines** [comment]
I know this is closed / resolved. However I too had issues with port audio under M1 Macs with macOS 12.6 and SDK 12.3.
At first I patched the macosx.patch to  support macosx12.3 SDK  which helped to build the app DemoHelloWorld. But then on the console , when starting the app, you would see 
`||PaMacCore (AUHAL)|| AUHAL component not found.` 
and pressing one of the buttons in the buttons in the app would not create any sound.

So I figured maybe a newer version of port audio would work better, which did the job. I altered the `make.sh` of port audio to the latest stable release of port audio and disabled all patching:
```
PKGURL=https://github.com/PortAudio/portaudio/archive/refs/tags/v19.7.0.tar.gz
PKGHASH=90676f9b7856bf100b396d8f14cc95bbfb91fa40

package_download $PKGURL $PKGHASH

# Restrict patching to two platforms
#if [ $SYS_HOSTPLATFORM = openbsd ] || [ $SYS_HOSTPLATFORM = macosx ]; then
#  package_patch
#fi
...
....
.....
```

**Comment by mgorges** [comment]
Merged new version of LIBPORTAUDIO in bfa1752. I hope the removal for OpenBSD patching works as the status of https://github.com/PortAudio/portaudio/issues/220 is unclear to me and I can't test for that platform.
